avocado/utils/build.py

```diff
@@ -33,17 +33,18 @@ def run_make(path, make='make', extra_args='', process_kwargs=None):
     os.chdir(path)
     cmd = make
 
+    if process_kwargs is None:
+        process_kwargs = {}
     env = {}
     # Set default number of jobs as ncpus + 1
-    if "-j" not in os.environ.get("MAKEFLAGS", ""):
+    makeflags = os.environ.get('MAKEFLAGS', '')
+    if process_kwargs.get('env') is None:
+        process_kwargs['env'] = {}
+    env_makeflags = process_kwargs['env'].get('MAKEFLAGS', '')
```

Comment:
... `env_makeflags` to `args_makeflags`, so variable names will be more related to the actual content.